### ISSUE TITLE
Numeric pins break everything

### ISSUE THREAD

**Issue by richo-stripe** [hwchen/keyring-rs#2]
On OSX, I have a numeric pin stored that I want to retrieve.

I'm assuming that the code here: https://github.com/hwchen/keyring-rs/blob/master/src/macos.rs#L98-L104 is meant to deal with binary objects (although I can't find any code for trying to serialize as hex), in the case of my ascii numeric pin it's trying to decode it as hex and then turn that into UTF 8 which fails, and as a result I can't fetch my entry.

Maybe this should be a different method? I can open a PR for that if you'd prefer.

**Comment by hwchen** [comment]
Hi, thanks for filing the issue. Sorry about the bug, I'll take a look at your PR.

**Comment by hwchen** [comment]
Hello,

Just need to verify how this (#3) is working for you (I don't currently have access to osx).

From my research, it looks like because I pass the `-w` flag to the `security` command, the password should come back as hex. As mentioned above, I can't currently verify this because I don't have access to osx. But the stackoverflow here mentions it in the expansion to the accepted answer: http://stackoverflow.com/questions/22370552/keychain-services-secure-notes

I'm pretty sure that's why the code is the way it is. Since it's been awhile since I looked at this code, it's possible I did something stupid. I had tested manually, I should get some stuff set up.

Also, hopefully I'll soon move the osx backend to use the API instead of the cli.
